Boost 1.83 survey

liushuyu opened the pull request:
Topic Description
-----------------

Boost 1.83 survey

Package(s) Affected
-------------------

```
runtime-common/boost
app-games/0ad
app-multimedia/aegisub
app-admin/aptitude
app-devel/binutils+cross-mips64r6el
runtime-display/imath
runtime-imaging/openexr
runtime-common/openvdb
runtime-imaging/openimageio
app-devel/cuda
app-creativity/blender
app-games/citra
app-devel/cpprestsdk
runtime-i18n/libime
app-i18n/fcitx5-chinese-addons
runtime-imaging/pcl
app-creativity/freecad
runtime-editors/source-highlight
app-devel/gdb+cross-arm64
app-devel/gdb+cross-mips64r6el
app-utils/gpick
app-devel/heaptrack
app-creativity/hugin
app-web/i2pd
app-creativity/inkscape
app-utils/innoextract
runtime-scientific/ixion
runtime-productivity/libkolabxml
desktop-kde/kdepim-runtime
desktop-kde/kig
runtime-desktop/libappimage
runtime-productivity/libcmis
runtime-productivity/liborcus
runtime-i18n/libphonenumber
runtime-productivity/libpst
runtime-i18n/librime
runtime-web/libtorrent-rasterbar
runtime-ham/libuhd
runtime-common/lucene++
app-multimedia/mkvtoolnix
app-games/mupen64plus
app-multimedia/ncmpcpp
app-electronics/prjtrellis
app-electronics/nextpnr
runtime-web/nghttp2
runtime-scientific/openbabel
app-games/openmw
extra-scientific/cgal
app-creativity/openscad
app-multimedia/pamixer
app-i18n/poedit
app-imaging/povray
runtime-common/process-cpp
app-electronics/pulseview
lang-python/pycuda
lang-python/pygattlib
app-games/rlvm
app-admin/snapper
app-multimedia/sunshine
app-creativity/superslicer
app-games/supertux
desktop-trinity/tdeedu
app-admin/thin-provisioning-tools
app-admin/thunderbolt-software-user-space
app-network/trojan
runtime-scientific/vigra
app-games/wesnoth
app-games/yuzu
app-multimedia/zxtune
```

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

liushuyu committed:
boost: update to 1.83.0

liushuyu committed:
boost: re-enable LTO

liushuyu committed:
cgal: re-introduce ...

... seems to be dropped by mistake

This reverts commit 7a0b011a9607f18a87aca903aac2e56fc0a2b6db.

liushuyu committed:
groups/boost-rebuild-minimal: add a new list ...

... for minimal boost rebuild

liushuyu committed:
0ad: rebuild for boost update

liushuyu committed:
boost: rebuild for boost update

liushuyu committed:
aegisub: rebuild for boost update

liushuyu committed:
binutils+cross-mips64r6el: rebuild for boost update

liushuyu committed:
aptitude: rebuild for boost update

liushuyu committed:
openexr: rebuild for boost update

liushuyu committed:
openvdb: rebuild for boost update

liushuyu committed:
openimageio: rebuild for boost update

liushuyu committed:
imath: rebuild for boost update

liushuyu committed:
cuda: rebuild for boost update

liushuyu committed:
fcitx5-chinese-addons: rebuild for boost update

liushuyu committed:
libime: rebuild for boost update

liushuyu committed:
citra: rebuild for boost update

liushuyu committed:
blender: rebuild for boost update

liushuyu committed:
cpprestsdk: rebuild for boost update

liushuyu committed:
gdb+cross-mips64r6el: rebuild for boost update

liushuyu committed:
source-highlight: rebuild for boost update

liushuyu committed:
gdb+cross-arm64: rebuild for boost update

liushuyu committed:
gpick: rebuild for boost update

liushuyu committed:
pcl: rebuild for boost update

liushuyu committed:
inkscape: rebuild for boost update

liushuyu committed:
innoextract: rebuild for boost update

liushuyu committed:
i2pd: rebuild for boost update

liushuyu committed:
hugin: rebuild for boost update

liushuyu committed:
heaptrack: rebuild for boost update

liushuyu committed:
kdepim-runtime: rebuild for boost update

liushuyu committed:
kig: rebuild for boost update

liushuyu committed:
libkolabxml: rebuild for boost update

liushuyu committed:
ixion: rebuild for boost update

liushuyu committed:
librime: rebuild for boost update

liushuyu committed:
libphonenumber: rebuild for boost update

liushuyu committed:
libpst: rebuild for boost update

liushuyu committed:
libtorrent-rasterbar: rebuild for boost update

liushuyu committed:
lucene++: rebuild for boost update

liushuyu committed:
mkvtoolnix: rebuild for boost update

liushuyu committed:
libuhd: rebuild for boost update

liushuyu committed:
mupen64plus: rebuild for boost update

liushuyu committed:
ncmpcpp: rebuild for boost update

liushuyu committed:
prjtrellis: rebuild for boost update

liushuyu committed:
openmw: rebuild for boost update

liushuyu committed:
nextpnr: rebuild for boost update

liushuyu committed:
cgal: rebuild for boost update

liushuyu committed:
nghttp2: rebuild for boost update

liushuyu committed:
openbabel: rebuild for boost update

liushuyu committed:
openscad: rebuild for boost update

liushuyu committed:
process-cpp: rebuild for boost update

liushuyu committed:
pamixer: rebuild for boost update

liushuyu committed:
poedit: rebuild for boost update

liushuyu committed:
povray: rebuild for boost update

liushuyu committed:
pulseview: rebuild for boost update

liushuyu committed:
sunshine: rebuild for boost update

liushuyu committed:
pygattlib: rebuild for boost update

liushuyu committed:
snapper: rebuild for boost update

liushuyu committed:
pycuda: rebuild for boost update

liushuyu committed:
rlvm: rebuild for boost update

liushuyu committed:
superslicer: rebuild for boost update

liushuyu committed:
thin-provisioning-tools: rebuild for boost update

liushuyu committed:
tdeedu: rebuild for boost update

liushuyu committed:
supertux: rebuild for boost update

liushuyu committed:
thunderbolt-software-user-space: rebuild for boost update

liushuyu committed:
trojan: rebuild for boost update

liushuyu committed:
zxtune: rebuild for boost update

liushuyu committed:
wesnoth: rebuild for boost update

liushuyu committed:
yuzu: rebuild for boost update

liushuyu committed:
vigra: rebuild for boost update

liushuyu committed:
boost: remove obsoleted patches

liushuyu committed:
boost: remove Python 2 bindings ...

... since it's not supported anymore

liushuyu committed:
boost: multiple changes ...

* set C++ standard to 17
* add xz to the dependency
* use $ABTHREADS instead of reading from $MAKEFLAGS

liushuyu committed:
boost: fix CFLAGS and CXXFLAGS settings

liushuyu committed:
boost: fix job option

liushuyu committed:
boost: fix the config file passing via Jamfile

liushuyu committed:
boost: reset REL to 0

liushuyu committed:
boost: add cxxflags for bootstrapped b2 tool

liushuyu committed:
boost: multiple changes ...

* build a full-featured b2 with the correct ab3 flags
* use the correct b2 binary to build the rest of boost

liushuyu committed:
0ad: add a patch to fix build with GCC 13

liushuyu committed:
cwidget: fix build with newer glibc

liushuyu committed:
cuda: update to 12.2.2+535.104.05

liushuyu committed:
blender: update to 3.6.4

liushuyu committed:
citra: update to nightly-2006

liushuyu committed:
heaptrack: update to 1.5.0

liushuyu committed:
pycuda: update to 2022.2.2

liushuyu committed:
libappimage: add upstream patches to fix build with GCC 13

jiegec commented:
Please apply https://github.com/boostorg/context/commit/819c2d6423b4e0f55c5f69e334bf81570e82f68f for loongarch64/riscv64 platforms, otherwise Boost.Context won't work.

liushuyu committed:
boost: add a patch from upstream to fix build ...

... with Tier-2 and 3 architectures

liushuyu commented:
> Please apply [boostorg/context@819c2d6](https://github.com/boostorg/context/commit/819c2d6423b4e0f55c5f69e334bf81570e82f68f) for loongarch64/riscv64 platforms, otherwise Boost.Context won't work.

Done in [c2c8c25](https://github.com/AOSC-Dev/aosc-os-abbs/pull/4759/commits/c2c8c25c578f3eb675d853d9ebe787bfccd4cba3)

liushuyu committed:
librime: update to 1.9.0

liushuyu committed:
openimageio: refresh checksums

liushuyu committed:
libuhd: update to 4.5.0.0

liushuyu committed:
mkvtoolnix: update to 79.0

liushuyu committed:
nextpnr: update to 0.6

liushuyu committed:
nextpnr: remove obsoleted patches

liushuyu committed:
openbabel: update to 3.1.1

liushuyu committed:
kalzium: rebuild for openbable update

liushuyu committed:
openmw: update to 0.48.0

liushuyu committed:
openmw: remove obsoleted patches

liushuyu committed:
openmw: add yaml-cpp dependency

liushuyu committed:
mygui: update to 3.4.3

liushuyu committed:
cgal: use header-only mode ...

... and mark as noarch

liushuyu committed:
rlvm: update to 0.14+git20220404

liushuyu committed:
rlvm: add a patch to fix build with GCC 13

liushuyu committed:
superslicer: update to 2.5.59.2

liushuyu committed:
supertux: update to 0.6.3

liushuyu committed:
supertux: add a patch to fix build with GCC 13

liushuyu committed:
wesnoth: update to 1.16.10

liushuyu committed:
wesnoth: add an upstream patch to fix build with boost 1.83

liushuyu committed:
yuzu: update to 1582

liushuyu committed:
zxtune: update to r5053

liushuyu committed:
zxtune: remove obsoleted patches

liushuyu committed:
libkolabxml: use Debian's source to avoid upstream server errors

liushuyu committed:
pycuda: update checksums and enable ARM64 target

liushuyu committed:
boost: remove Boost.Python Python 2 references

liushuyu committed:
boost: fix the file path in the 0003 patch

liushuyu committed:
cgal: move the directory to the correct category

liushuyu committed:
openbabel: fix the CMake reference in the CMake modules

liushuyu committed:
aegisub: add a patch to fix build with Boost 1.83

liushuyu committed:
aegisub: correct dependency information

liushuyu committed:
gpick: update to 0.3

liushuyu committed:
gpick: switch to use CMake

liushuyu committed:
nextpnr: use a fixed-version of mistral

liushuyu committed:
prjtrellis: update to 1.4

liushuyu committed:
apicula: update to 0.9.1

liushuyu committed:
apicula: extract more device prebuilt files

liushuyu committed:
libva-nvidia-driver: update to 0.0.11

liushuyu committed:
openmw: use bundled MyGUI for compatibility

liushuyu committed:
openscad: update to 2021.01+git20230825

liushuyu committed:
superslicer: refresh patches

liushuyu committed:
groups/boost-rebuild-20231014: add a rebuild list for boost 1.83

liushuyu committed:
gloox: fix build on GCC 13

liushuyu committed:
gloox: update to 1.0.27

liushuyu committed:
groups/boost-rebuild-20231014: add gloox

liushuyu committed:
0ad: unset RUSTFLAGS for SpiderMonkey

liushuyu committed:
openvdb: use release build type ...

... to workaround a tbb CMake issue

liushuyu committed:
blender: add a hack to force NVCC to use GCC 13

liushuyu committed:
groups/boost-rebuild-20231014: fix cgal path

liushuyu committed:
mupen64plus: enable x86 SIMD translation on ppc64el

liushuyu committed:
zxtune: avoid stripping by null'ing the objdump and strip commands

liushuyu committed:
aptitude: re-enable RECONF and remove hacks

liushuyu committed:
libsigrok: gate -mxgot flag behind MIPS architecture

liushuyu committed:
pandas: update to 2.1.2

liushuyu committed:
numpy: update to 1.26.1 ...

... also switch to Meson

liushuyu committed:
meson: update to 1.2.3

liushuyu committed:
numpy: use the Meson wrapper in the tree

liushuyu committed:
numpy: use Clang ...

... upstream now uses some Clang/LLVM intrinsics in the source

liushuyu committed:
numpy: use lld if possible

liushuyu committed:
versioneer: new, 0.29 ...

... required by pandas

liushuyu committed:
pandas: add meson and versioneer to the build deps

liushuyu committed:
numpy: add openblas as the dependency

liushuyu committed:
pandas: add cython to the build deps

liushuyu committed:
process-cpp: switch to the correct upstream

liushuyu committed:
properties-cpp: update to 0.0.3

liushuyu committed:
properties-cpp: add a patch to remove coverage collection

liushuyu committed:
properties-cpp: mark as noarch ...

... this is a header-only library

liushuyu committed:
potrace: replace x86 assembly with compiler intrinsics

liushuyu committed:
potrace: bump REL for patch additions

liushuyu committed:
libsigrok: add a patch to fix SWIG 4 binding generation

liushuyu committed:
prjoxide: update to 0+20230831

liushuyu committed:
prjoxide: disable LTO on LoongArch64

liushuyu committed:
prjoxide: generate the lockfile

liushuyu committed:
sdl-ttf: add a patch to check FreeType2 using pkg-config ...

... instead of using freetype-config

liushuyu committed:
physfs: lint build scripts ...

... and also fix build after using Ninja

liushuyu committed:
automatic-component-toolkit: disable ...

... Go PIE executable generation on loongarch64, also making the build
output more verbose

liushuyu committed:
gtk-3: migrate to Meson

liushuyu committed:
numpy: gate lld behind USECLANG

liushuyu committed:
numpy: disable clang for loongson3

MingcongBai commented on app-creativity/blender/autobuild/patches/HACK-impersonate-nvcc-gcc-13.patch.deferred:
Fix up the file name here.

liushuyu replied:
This is done on purpose. This patch needs to be applied out-of-sequence.

MingcongBai commented on app-electronics/apicula/autobuild/prepare:
```diff
@@ -1,5 +1,8 @@
+	 apycula/GW1NZ-1.pickle apycula/GW2A-18.pickle apycula/GW2A-18C.pickle)
+for i in "${DEVICES[@]}"
 do
 	unzip ../apycula.whl $i
```
Absolute path here.

MingcongBai commented on app-creativity/superslicer/autobuild/patches/superslicer-backport-to-tbb-2020.patch:
Fix up file name here.

MingcongBai commented on app-electronics/nextpnr/spec:
```diff
@@ -1,7 +1,7 @@
-SRCS="git::commit=b7207b088539e7b5f34ebbd9f53f9fa14bef6e62;rename=nextpnr::https://github.com/YosysHQ/nextpnr.git \
-      git::commit=7c8ba15beb174d8def021bba6e1db06426e318a9;rename=mistral::https://github.com/Ravenslofty/mistral"
+VER=0.6
+# NOTE: nextpnr requires a specific version of mistral
```
Please add a comment on how to track mistral versions. Also, consider adding mistral version to the main VER.

liushuyu replied:
> Please add a comment on how to track mistral versions. Also, consider adding mistral version to the main VER.

I don't know how to track mistral versions. This is purely trial and error.

MingcongBai commented on app-electronics/prjoxide/autobuild/defines:
```diff
@@ -5,4 +5,5 @@ BUILDDEP="llvm rustc"
 PKGDES="Libraries and utilities for open-source toolchain of Lattice Nexus series FPGAs"
 
 USECLANG=1
+NOLTO__LOONGARCH64=1
```
Add a FIXME comment here.

MingcongBai commented on app-games/0ad/autobuild/build:
```diff
@@ -1,6 +1,7 @@
 (
     abinfo "Configuring $PKGNAME ..."
     cd "$SRCDIR"/build/workspaces
+    unset RUSTFLAGS
```
Please add a FIXME comment here explaining why.

MingcongBai commented on app-games/mupen64plus/autobuild/build:
```diff
@@ -12,6 +12,11 @@ if [[ "${CROSS:-$ARCH}" = "armv7hf" || \
     NEON="NEON=1 VFP_HARD=1"
 fi
 
+if [[ "${CROSS:-$ARCH}" = "ppc64el" ]]; then
```
Please use `ab_match_arch`.

MingcongBai commented on app-multimedia/zxtune/autobuild/build:
```diff
@@ -1,30 +1,24 @@
-    CXX_MODE_FLAGS="${CPPFLAGS} ${CXXFLAGS} -fPIC -DNDEBUG" \
-    LD_MODE_FLAGS="${LDFLAGS}" \
-    -C apps/bundle/
+make "${OPTIONS[@]}" -C apps/bundle/
```
Absolute path here.

MingcongBai commented on app-multimedia/zxtune/autobuild/build:
```diff
@@ -1,30 +1,24 @@
-    -C apps/bundle/
+     "${OPTIONS[@]}" \
+     DESTDIR="$PKGDIR" \
+     -C apps/bundle/
```
Ditto.

MingcongBai commented on desktop-gnome/gtk-3/autobuild/defines:
Rebase against `stable`. This change was cherry-picked out.

MingcongBai commented on groups/boost-rebuild-minimal:
Please add a comment explaining what this is used for.

MingcongBai commented on runtime-common/openvdb/autobuild/defines:
```diff
@@ -6,12 +6,14 @@ BUILDDEP="doxygen llvm texlive"
 
 #FIXME: OpenVDB AX complains about too new LLVM (15+)
 #       Setting -DUSE_AX=OFF for now
+#FIXME: Due to tbb CMake configurations problem
```
Can we fix tbb?

liushuyu replied:
Not in this topic. It would require a separate rebuild

MingcongBai commented on runtime-common/physfs/autobuild/beyond:
```diff
@@ -1,6 +1,9 @@
-install -m644 docs/html/* "$PKGDIR"/usr/share/doc/physfs
-install -m644 docs/man/man3/* "$PKGDIR"/usr/share/man/man3
+abinfo "Generating documentation ..."
+cmake --build "$SRCDIR" -- docs
```
Absolute path here.

liushuyu replied:
`docs` is a target name, not a path.

MingcongBai commented on runtime-common/physfs/autobuild/beyond:
```diff
@@ -1,6 +1,9 @@
+
+install -dv "$PKGDIR"/usr/share/{doc/physfs,man/man3}
+install -Dvm644 docs/html/* "$PKGDIR"/usr/share/doc/physfs
+install -Dvm644 docs/man/man3/* "$PKGDIR"/usr/share/man/man3
```
Ditto.

MingcongBai commented on runtime-display/nvidia/spec:
Rebase against stable, as @cthbleachbit updated NVIDIA recently for Kernel 6.6

MingcongBai commented on runtime-i18n/librime/spec:
```diff
@@ -1,15 +1,14 @@
-LUACOMMIT=8b37d5541b0341a07ed7517c78df1a09268001fd
-CHARCODECOMMIT=b569184772b12965e3ebe1dfd431026951fed81c
+VER=1.9.0
+OCTAGRAMCOMMIT=a6ced5a1d623afed17284fbaa395dd3a8b019a14
```
Please add a comment explaining how we should track the commits.

MingcongBai replied:
Also, I wonder why they were bundled in the first place...

MingcongBai commented on runtime-multimedia/libva-nvidia-driver/spec:
Please rebase against stable, @RedL0tus updated this package in the Graphical Stack Survey (20231031).

MingcongBai commented on runtime-scientific/openbabel/autobuild/beyond:
```diff
@@ -0,0 +1,3 @@
+abinfo "Moving CMake files to the correct location ..."
+cp -rv "$PKGDIR"/lib "$PKGDIR"/usr/
+rm -r "$PKGDIR"/lib
```
Verbosity here.

MingcongBai commented on runtime-scientific/openbabel/autobuild/beyond:
Please see if this is still needed with our recent `bash-startup` update, which now favours /usr/bin over /bin.

liushuyu committed:
libappimage: rebuild for boost update

liushuyu committed:
liborcus: rebuild for boost update

liushuyu committed:
libcmis: rebuild for boost update

liushuyu committed:
cwidget: remove duplicated patch

liushuyu committed:
citra: update to 2041

liushuyu committed:
prjoxide: add FIXME comment on LA64 LTO

liushuyu committed:
0ad: add a comment on "unset RUSTFLAGS" part

liushuyu committed:
mupen64plus: use ab_match_arch to replace legacy architecture detection

liushuyu committed:
zxtune: use absolute path for make

liushuyu committed:
automatic-component-toolkit: lint build script

liushuyu committed:
groups/boost-rebuild: renamed from groups/boost-rebuild-minimal

liushuyu committed:
physfs: use absolute path for install

liushuyu committed:
libsigrok: remove -mxgot workaround ...

... as this is now included in ab3

liushuyu committed:
librime: add a note for subproject commits

liushuyu committed:
openbabel: add verbosity for beyond script

liushuyu committed:
boost: enable python.numpy support
